src/components/shared/Entity/Actions/Save.tsx

```diff
@@ -56,35 +54,45 @@ function EntityCreateSaveButton({ disabled, formId, onFailure }: Props) {
     const resetFormState = useEntityCreateStore(
         entityCreateStoreSelectors.formState.reset
     );
-    const formStateStatus = useEntityCreateStore(
-        entityCreateStoreSelectors.formState.status
-    );
     const messagePrefix = useEntityCreateStore(
         entityCreateStoreSelectors.messagePrefix
     );
+    const formActive = useEntityCreateStore(
+        entityCreateStoreSelectors.isActive
+    );
 
-    const waitForPublishToFinish = (logToken: string) => {
-        resetFormState(FormStatus.SAVING);
+    const waitForPublishToFinish = (logTokenVal: string) => {
+        resetFormState(status);
         console.log('wait for finish');
         const subscription = startSubscription(
             supabaseClient.from(
                 `${TABLES.PUBLICATIONS}:draft_id=eq.${draftId}`
             ),
             async (payload: any) => {
-                if (payload.logs_token === logToken) {
+                if (payload.logs_token === logTokenVal) {
                     setPubId(payload.id);
                     setFormState({
-                        status: FormStatus.SUCCESS,
-                        exitWhenLogsClose: true,
+                        status: dryRun ? FormStatus.TESTED : FormStatus.SAVED,
+                        exitWhenLogsClose: !dryRun,
                     });
 
+                    let description, title;
+
+                    if (!dryRun) {
+                        description = `${messagePrefix}.createNotification.desc`;
+                        title = `${messagePrefix}.createNotification.title`;
+                    } else {
+                        description = `${messagePrefix}.testNotification.desc`;
+                        title = `${messagePrefix}.testNotification.title`;
+                    }
```

Comment:
Save/Test buttons are the same thing but only differ if it is dry running or not. So now need to handle setting message.